src/Compilers/Core/Portable/DiagnosticAnalyzer/AnalyzerDriver.cs

```diff
@@ -742,38 +741,25 @@ private void ExecuteSyntaxTreeActions(AnalysisScope analysisScope, AnalysisState
                 var file = new SourceOrAdditionalFile(tree);
                 if (isGeneratedCode && DoNotAnalyzeGeneratedCode)
                 {
-                    analysisState?.TryMarkSyntaxAnalysisComplete(file, analysisScope.Analyzers);
                     continue;
                 }
 
-                var processedAnalyzers = analysisState != null ? PooledHashSet<DiagnosticAnalyzer>.GetInstance() : null;
-                try
+                foreach (var (analyzer, syntaxTreeActions) in _lazySyntaxTreeActions)
                 {
-                    foreach (var (analyzer, syntaxTreeActions) in _lazySyntaxTreeActions)
+                    if (!analysisScope.Contains(analyzer))
                     {
-                        if (!analysisScope.Contains(analyzer))
-                        {
-                            continue;
-                        }
-
-                        cancellationToken.ThrowIfCancellationRequested();
-
-                        // Execute actions for a given analyzer sequentially.
-                        AnalyzerExecutor.TryExecuteSyntaxTreeActions(syntaxTreeActions, analyzer, file, analysisScope, analysisState, isGeneratedCode);
-
-                        processedAnalyzers?.Add(analyzer);
+                        continue;
                     }
 
-                    analysisState?.TryMarkSyntaxAnalysisCompleteForUnprocessedAnalyzers(file, analysisScope, processedAnalyzers!);
-                }
-                finally
-                {
-                    processedAnalyzers?.Free();
+                    cancellationToken.ThrowIfCancellationRequested();
+
+                    // Execute actions for a given analyzer sequentially.
+                    AnalyzerExecutor.ExecuteSyntaxTreeActions(syntaxTreeActions, analyzer, file, isGeneratedCode);
```

Comment:
📝 Noticed that `analysisScope` is no longer passed here.

Reply:
➡️ This is because the only remaining use of `analysisScope` in the callee is an assertion that `analysisScope.Contains(analyzer)`, which we know from a couple lines above.